R/trackhub.R

```diff
@@ -250,11 +239,66 @@ setClass("TrackHubGenome",
 
 trackhub <- function(x) x@trackhub
 
+transformLogicalFields <- function(track) {
+    logicalFields <- c(
+    "boxedCfg", "darkerLabels", "skipEmptyFields", "indelDoubleInsert", "indelQueryInsert",
+    "indelPolyA", "showNames", "doWiggle", "itemRgb", "labelOnFeature", "exonArrows", "exonNumbers",
+    "noScoreFilter", "spectrum", "thickDrawItem", "exonArrowsDense", "mergeSpannedItems", "linkIdInName",
+    "showDiffBasesAllScales", "alwaysZero", "negateValues", "yLineOnOff", "gridDefault", "compositeTrack",
+    "allButtonPair", "centerLabelsDense", "hideEmptySubtracks", "viewUi", "configurable", "showSubtrackColorOnUi",
+    "noInherit", "hapClusterEnabled", "applyMinQual")
+    i <- 1L
+    totalFields <- length(logicalFields)
+    while(i <= totalFields) {
+        track <- sub(paste0(logicalFields[i] ,"= '(on|On)'"), paste0(logicalFields[i], "=TRUE"), track)
+        track <- sub(paste0(logicalFields[i] ,"= '(off|Off)'"), paste0(logicalFields[i], "=FALSE"), track)
+        track <- sub(paste0(logicalFields[i] ,"= '(true)'"), paste0(logicalFields[i], "=TRUE"), track)
+        track <- sub(paste0(logicalFields[i] ,"= '(false)'"), paste0(logicalFields[i], "=FALSE"), track)
+        i <- i + 1
+    }
+    track
+}
+
+transformIntegerFields <- function(track) {
+    integerFields <- c("maxWindowToDraw", "denseCoverage", "maxItems", "scoreMax", "scoreMin", "maxWindowToQuery",
+                       "showSnpWidth", "useScore")
+    i <- 1L
+    totalFields <- length(integerFields)
+    while(i <= totalFields) {
+        value <- regmatches(track, regexec(paste0(integerFields[i], "= '[0-9]*"), track))
+        value <- sub(paste0(integerFields[i], "= '"), "as.integer(", value)
+        track <- sub(paste0(integerFields[i], "= '[0-9]+'"), paste0(integerFields[i], "=", value, ")"), track)
+        i <- i + 1
+    }
+    track
+}
+
+transformNumericFields <- function(track) {
+    numericFields <- c("priority")
+
+    i <- 1L
+    totalFields <- length(numericFields)
+    while(i <= totalFields) {
+        value <- regmatches(track, regexec(paste0(numericFields[i], "= '[0-9]*"), track))
+        value <- sub(paste0(numericFields[i], "= '"), "as.numeric(", value)
+        track <- sub(paste0(numericFields[i], "= '[0-9]+'"), paste0(numericFields[i], "=", value, ")"), track)
+        i <- i + 1
+    }
+    track
+}
+
+transformTracksFields <- function(track) {
```

Comment:
Instead of converting the file to R code, please consider a simpler approach that treats the file as data. For example, if the fields are `field=value`, it should be easy to get them into a data.frame using `strcapture()`. You can have a "schema" list that maps the field name to data type. Like `fieldToType <- list(boxedCfg="logical", maxWindowToDraw="integer", ...)`. Then you can do something like `args <- Map(as, df$value, fieldToType[df$field])` to perform the conversion. Although `as()` is not quite smart enough, for example, it won't convert "on" to `TRUE`. You could fix that in a preprocessing step using something like `sub("'[Oo]n'", "true", df$value)`. Then you can pass those args to `Track()` using `do.call(Track, args)`. Just a sketch.